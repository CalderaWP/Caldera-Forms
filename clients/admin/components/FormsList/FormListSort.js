import React from 'react';
import PropTypes from 'prop-types'
import {RenderGroup} from '@caldera-labs/components';

export const SORT_FORMS_BY_NAME = 'name';
export const SORT_FORMS_BY_UPDATE = '_last_updated';

/**
 * Form list sort and search controls.
 *
 * @param props
 * @return {*}
 * @constructor
 */
export const FormListSort = (props) => {

	const sortOptions = {
		id: 'cf-form-sort-options',
		label: 'Sort Forms By',
		desc: '',
		type: 'select',
		inputClass: FormListSort.classNames.order,
		value: props.order,
		options: [
			{
				value: SORT_FORMS_BY_NAME,
				label: 'Name'
			},
			{
				value: SORT_FORMS_BY_UPDATE,
				label: 'Last Update'
			}
		],
		onValueChange: (event) => {
			props.onChangeOrder(event.target.value)

		}
	};
	const searchField = {
		id: 'cf-form-search',
		label: 'Search Forms By',
		desc: '',
		type: 'input',
		innerType: 'search',
		inputClass: FormListSort.classNames.search,
		value: props.formSearchTerm,
		onValueChange: (newValue) => {
			props.onFormSearch(newValue);
		}
	};
	return (
		<div>
			<RenderGroup
				configFields={[
					searchField,
					sortOptions
				]}
				className={'caldera-forms-admin-list-sort'}
			/>
		</div>
	);
};


FormListSort.propTypes = {
	order: PropTypes.oneOf([
		SORT_FORMS_BY_NAME,
		SORT_FORMS_BY_UPDATE
	]),
	onChangeOrder: PropTypes.func,
	formSearchTerm: PropTypes.string,
	onFormSearch: PropTypes.func

};

FormListSort.defaultProps = {
	order: SORT_FORMS_BY_NAME
};

FormListSort.classNames = {
	order: 'order-forms-select',
	search: 'search-froms'
};


