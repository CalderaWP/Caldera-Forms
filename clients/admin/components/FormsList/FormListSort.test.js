import renderer from "react-test-renderer";
import {FormListSort,SORT_FORMS_BY_NAME, SORT_FORMS_BY_UPDATE} from "./FormListSort";
import React from 'react';
import {mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

describe('FormListSort component', () => {
	it('Matches snapshot', () => {
		const formList = renderer.create(
			<FormListSort
				onChangeOrder={() => {} }
				order={SORT_FORMS_BY_NAME}
			/>
		);
		expect(formList.toJSON()).toMatchSnapshot();
	});

	it( 'updates order', () => {
		let recived = '';
		const component = mount(
			<FormListSort
				onChangeOrder={(
					newValue
				) => {
					recived = newValue;
				}
				}
				order={SORT_FORMS_BY_NAME}
			/>
		);

		component.find( '.' + FormListSort.classNames.order ).simulate('change',
			{target: { value: SORT_FORMS_BY_UPDATE}}
		);
		expect(recived).toEqual(SORT_FORMS_BY_UPDATE);
	});




});