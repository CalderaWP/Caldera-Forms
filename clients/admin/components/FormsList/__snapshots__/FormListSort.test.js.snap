// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormListSort component Matches snapshot 1`] = `
<div>
  <div
    className="caldera-forms-admin-list-sort"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="cf-form-sort-options"
        >
          Sort Forms By
        </label>
        <select
          className="field-config field-config"
          disabled={false}
          id="cf-form-sort-options"
          onChange={[Function]}
          value="name"
        >
          <option
            className="caldera-config-option"
            value="name"
          >
            Name
          </option>
          <option
            className="caldera-config-option"
            value="_last_updated"
          >
            Last Update
          </option>
        </select>
        
      </div>
    </div>
  </div>
</div>
`;
